test/extended/kubevirt/util.go

```diff
@@ -0,0 +1,317 @@
+		}
+	}
+	return "", fmt.Errorf("no internal node ip found for node %s", node.Name)
+}
+
+func checkKubeVirtGuestClusterHostNetworkNodePortConnectivity(serverFramework, clientFramework *e2e.Framework) error {
+	return checkKubeVirtGuestClusterConnectivity(serverFramework, clientFramework, true, v1.ServiceTypeNodePort)
+}
+
+func checkKubeVirtGuestClusterPodNetworkNodePortConnectivity(serverFramework, clientFramework *e2e.Framework) error {
+	return checkKubeVirtGuestClusterConnectivity(serverFramework, clientFramework, false, v1.ServiceTypeNodePort)
+}
+
+func checkKubeVirtInfraClusterConnectivity(serverFramework, clientFramework *e2e.Framework, oc *exutil.CLI, serviceType v1.ServiceType) error {
+	networking.MakeNamespaceScheduleToAllNodes(serverFramework)
+
+	serverGuestNode, err := getSchedulableNode(serverFramework)
+	Expect(err).To(Succeed())
+
+	serverVMPod, err := getKubeVirtPodFromGuestNode(clientFramework, serverGuestNode)
+	Expect(err).To(Succeed())
+
+	serverInfraNode := serverVMPod.Spec.NodeName
+	serverGuestNodeIP := serverVMPod.Status.PodIP
+
+	infraClientNode, err := getSchedulableInfraNode(clientFramework, serverInfraNode)
+	Expect(err).To(Succeed())
+
+	podName := names.SimpleNameGenerator.GenerateName("service-")
+	defer serverFramework.ClientSet.CoreV1().Pods(serverFramework.Namespace.Name).Delete(context.Background(), podName, metav1.DeleteOptions{})
+	defer serverFramework.ClientSet.CoreV1().Services(serverFramework.Namespace.Name).Delete(context.Background(), podName, metav1.DeleteOptions{})
+
+	serviceAddr := ""
+	if serviceType == v1.ServiceTypeNodePort {
+		serverGuestNodePort := launchWebserverNodePortService(serverFramework.ClientSet, serverFramework.Namespace.Name, podName, serverGuestNode)
+		serviceAddr = net.JoinHostPort(serverGuestNodeIP, strconv.Itoa(int(serverGuestNodePort)))
+	}
+	if serviceType == v1.ServiceTypeLoadBalancer {
+		externalIP := strings.TrimLeft(strings.Split(serverFramework.ClientConfig().Host, ":")[1], "/")
+		serverLBPort := launchWebserverLoadBalancerService(serverFramework.ClientSet, serverFramework.Namespace.Name, podName, serverGuestNode, []string{externalIP})
+		serviceAddr = net.JoinHostPort(externalIP, strconv.Itoa(int(serverLBPort)))
+	}
+
+	return checkConnectivityToHostWithCLI(clientFramework, oc, infraClientNode, "service-wget", serviceAddr, 10*time.Second, false)
+}
+
+func checkKubeVirtInfraClusterNodePortConnectivity(serverFramework, clientFramework *e2e.Framework, oc *exutil.CLI) error {
+	return checkKubeVirtInfraClusterConnectivity(serverFramework, clientFramework, oc, v1.ServiceTypeNodePort)
+}
+
+func checkKubeVirtGuestClusterConnectivity(serverFramework, clientFramework *e2e.Framework, hostNetwork bool, serviceType v1.ServiceType) error {
+	networking.MakeNamespaceScheduleToAllNodes(serverFramework)
+	networking.MakeNamespaceScheduleToAllNodes(clientFramework)
+	serverNode, clientNode, err := networking.FindAppropriateNodes(serverFramework, networking.DIFFERENT_NODE)
```

Comment:
Ugh, `findAppropriateNodes` is a disaster; it tries to strip master and infra nodes from the list, except that it uses the 3.x node label values rather than the 4.x ones so it totally fails. :joy: And so of course, then the tests that use `findAppropriateNodes` end up having to use `makeNamespaceScheduleToAllNodes` because they accidentally schedule random pods to the masters...

If you use `e2enode.GetBoundedReadySchedulableNodes(serverFramework.ClientSet, 2)`, you'll get back 2 nodes that don't include any tainted nodes (and thus don't include the masters), and then you don't need to call `makeNamespaceSchedulableToAllNodes` either. (You shouldn't need to call that in any other tests either.)